File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int area = 0;
        while (left < right)
        {
            int a = min(height[left], height[right]) * (right - left);
            area = max(area, a);
            if (height[left] < height[right])
            {
                left += 1;
            }
            else
            {
                right -= 1;
            }
        }
        return area;
    }
    
    public int min(int a, int b)
    {
        if (a < b)
        {
            return a;
        }
        return b;
    }
    
    public int max(int a, int b)
    {
        if (a > b)
        {
            return a;
        }
        return b;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        low = 0
        high = len(height) - 1
        max_water = 0
        while low < high:
            water = (high - low) * min(height[low], height[high])
            max_water = max(max_water, water)
            if height[low] <= height[high]:
                low += 1
            else:
                high -= 1
        return max_water